src/bench/addrman.cpp

```diff
@@ -130,10 +115,17 @@ static void AddrManGood(benchmark::Bench& bench)
         }
     };
 
-    uint64_t i = 0;
     bench.run([&] {
-        markSomeAsGood(*addrmans.at(i));
-        ++i;
+        // To make the benchmark independent of the number of evaluations, we always prepare a new addrman.
+        // This is necessary because CAddrMan::Good() method modifies the object, affecting the timing of subsequent calls
+        // to the same method and we want to do the same amount of work in every loop iteration.
+        //
+        // This has some overhead (about 8% on my machine), but that overhead is constant so improvements in
```

Comment:
Likely going to change with #22974

Reply:
hm, I don't know much about what addrman does, but currently `markSomeAsGood` in the benchmark only uses every 32nd element. Can be bumped up to call Good() on every entry, maybe in random order?

Reply:
I've rebased because #22974 is already merged, and updated the benchmark in 153e686 so it makes more sense with the much faster `Good()`. Nothing else has changed

Reply:
If the overhead is now 99%, then changes in the order of -50% or +100% are not noticed. Just asking to make sure the overhead isn't too much.

Reply:
This is what I get when I run `./src/bench/bench_bitcoin -min_time=10000 -filter="AddrManAdd.*"` on my machine:

|               ns/op |                op/s |    err% |          ins/op |          cyc/op |    IPC |         bra/op |   miss% |     total | benchmark
|--------------------:|--------------------:|--------:|----------------:|----------------:|-------:|---------------:|--------:|----------:|:----------
|       44,370,397.17 |               22.54 |    0.6% |  448,854,352.19 |  141,515,648.83 |  3.172 |  33,913,128.43 |    0.1% |     11.05 | `AddrManAdd`
|      114,389,015.11 |                8.74 |    0.2% |1,201,264,095.33 |  364,837,620.00 |  3.293 |  78,812,510.33 |    0.1% |     10.87 | `AddrManAddThenGood`

So with the change it's ~44ms for `AddrManAdd`, ~114ms for `AddrManAddThenGood`, leaving ~70ms for just the Good() part of the benchmark